azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/data/DbHelper.kt

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.data
+
+import android.content.Context
+import android.database.sqlite.SQLiteDatabase
+import android.database.sqlite.SQLiteOpenHelper
+import android.provider.BaseColumns
+
+internal class DbHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
+    override fun onCreate(db: SQLiteDatabase) {
+        db.execSQL(CallHistoryContract.SQL_CREATE_CALL_HISTORY)
+    }
+
+    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
+    }
+
+    override fun onDowngrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
+    }
+
+    companion object {
+        // If you change the database schema, you must increment the database version.
+        const val DATABASE_VERSION = 1
+        const val DATABASE_NAME = "com.azure.android.communication.ui.calling.CallHistoryReader.db"
+    }
+}
+
+internal object CallHistoryContract {
+
+    const val ID = BaseColumns._ID
+    const val TABLE_NAME = "call_history"
+
+    const val COLUMN_NAME_CALL_ID = "call_id"
+    const val COLUMN_NAME_CALL_DATE = "call_date"
+
+    const val SQL_CREATE_CALL_HISTORY =
+        "CREATE TABLE $TABLE_NAME (" +
```

Comment:
Do a `CREATE TABLE IF NOT EXISTS` just in case. E.g. if somehow `onCreate` is called while we already have a an existing database. There's a small time window when this can happen, I think - `SQLiteOpenHelper` constructor is called and it determines that there's no db on disk yet, and it needs to create one. So, it creates it, calls `onCreated`, but maybe before that can do its thing, OS kills the app.

Then you end up with an empty db file on disk, and next time this runs, we'll permafail - we have the db already, probably with the correct user_version pragma, so no need to call `oncreate`. But, that db is actually empty, without your table in it. And so everything breaks.

Or, maybe, this scenario isn't possible, and this is all moot. But I think it is possible. Either way, appending `IF NOT EXISTS` lets us to not worry about this :-)

Reply:
Add `IF NOT EXISTS`